pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -231,87 +236,461 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: secretServer.Name,
+										},
+										Items: []corev1.KeyToPath{
+											{
+												Key:  secretutils.DataKeyCertificate,
+												Path: secretutils.DataKeyCertificate,
+											},
+											{
+												Key:  secretutils.DataKeyPrivateKey,
+												Path: secretutils.DataKeyPrivateKey,
+											},
+										},
+									},
 								},
 							},
 						},
 					},
 				},
+				{
+					Name: volumeNameTLSAuth,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName:  secretTLSAuth.Name,
+							DefaultMode: pointer.Int32(0400),
+						},
+					},
+				},
+				{
+					Name: volumeNameDH,
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName:  dhSecret.Name,
+							DefaultMode: pointer.Int32(0400),
+						},
+					},
+				},
 			},
-			PolicyTypes: []networkingv1.PolicyType{
-				networkingv1.PolicyTypeIngress,
-				networkingv1.PolicyTypeEgress,
+		},
+	}
+
+	if !v.highAvailabilityEnabled {
+		template.Spec.Containers = append(template.Spec.Containers, corev1.Container{
+			Name:            envoyProxyContainerName,
+			Image:           v.imageAPIServerProxy,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			SecurityContext: &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{
+						"all",
+					},
+				},
+			},
+			Command: []string{
+				"envoy",
+				"--concurrency",
+				"2",
+				"-c",
+				fmt.Sprintf("%s/%s", volumeMountPathEnvoyConfig, fileNameEnvoyConfig),
+			},
+			ReadinessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					TCPSocket: &corev1.TCPSocketAction{
+						Port: intstr.FromInt(EnvoyPort),
+					},
+				},
+			},
+			LivenessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					TCPSocket: &corev1.TCPSocketAction{
+						Port: intstr.FromInt(EnvoyPort),
+					},
+				},
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("20m"),
+					corev1.ResourceMemory: resource.MustParse("20Mi"),
+				},
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: resource.MustParse("850M"),
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      volumeNameCerts,
+					MountPath: volumeMountPathCerts,
+				},
+				{
+					Name:      volumeNameEnvoyConfig,
+					MountPath: volumeMountPathEnvoyConfig,
+				},
 			},
+		})
+		template.Spec.Volumes = append(template.Spec.Volumes, corev1.Volume{
+			Name: volumeNameEnvoyConfig,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: configMap.Name,
+					},
+				},
+			},
+		})
+	} else {
+		mount := corev1.VolumeMount{
+			Name:      volumeNameStatusDir,
+			MountPath: volumeMountPathStatusDir,
```

Comment:
Drop this variable and just inline it below

Reply:
changed